src/modules/app/containers/Switcher.js

```diff
@@ -29,102 +24,88 @@ import { DISCLAIMER_ROUTE } from '../routes/disclaimer'
-import { NOT_FOUND } from 'redux-first-router'
+import Payload from '../../endpoint/Payload'
+import Failure from '../../common/components/Failure'
 
 type Props = {
   viewportSmall: boolean,
   currentRoute: string,
-  cities: Array<CityModel>,
-  categories: CategoriesMapModel,
-  events: Array<EventModel>,
-  extras: Array<ExtraModel>,
-  disclaimer: DisclaimerModel
+  citiesPayload: Payload,
+  categoriesPayload: Payload,
+  eventsPayload: Payload,
+  extrasPayload: Payload,
+  disclaimerPayload: Payload
 }
 
 /**
  * Renders different Pages depending on the current route. If the needed data is not available, a LoadingSpinner is rendered
  */
 class Switcher extends React.Component<Props> {
-  getComponent () {
-    const {currentRoute, cities, events, categories, extras, disclaimer} = this.props
+  static getDisplayComponent (page: React.Node, payload: Payload): React.Node {
     const LoadingSpinner = () => <Spinner name='line-scale-party' />
+    if (payload.isFetching) {
+      return <LoadingSpinner />
+    } else if (payload.data) {
+      return page
+    } else if (payload.error) {
+      return <Failure error={payload.error} />
+    } else {
+      return null
+    }
+  }
+
+  getPage (): React.Node {
+    const {currentRoute, citiesPayload, eventsPayload, categoriesPayload, extrasPayload, disclaimerPayload} = this.props
 
     switch (currentRoute) {
-      case NOT_FOUND:
-        if (cities) {
-          return <NotFoundPage />
-        }
-        break
       case LANDING_ROUTE:
-        if (cities) {
-          return <LandingPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<LandingPage />, citiesPayload)
       case MAIN_DISCLAIMER_ROUTE:
         return <MainDisclaimerPage />
       case CATEGORIES_ROUTE:
-        if (categories) {
-          return <CategoriesPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<CategoriesPage />, categoriesPayload)
       case EVENTS_ROUTE:
-        if (events) {
-          return <EventsPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<EventsPage />, eventsPayload)
       case EXTRAS_ROUTE:
-        if (extras) {
-          return <ExtrasPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<ExtrasPage />, extrasPayload)
       case DISCLAIMER_ROUTE:
-        if (disclaimer) {
-          return <DisclaimerPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<DisclaimerPage />, disclaimerPayload)
       case SEARCH_ROUTE:
-        if (categories) {
-          return <SearchPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<SearchPage />, categoriesPayload)
       case PDF_FETCHER_ROUTE:
-        if (categories) {
-          return <PdfFetcherPage />
-        }
-        break
+        return Switcher.getDisplayComponent(<PdfFetcherPage />, categoriesPayload)
     }
-    return <LoadingSpinner />
+    return null
```

Comment:
We could think about displaying an error here 🤔, I would also prefer to put that into a `default` case

Reply:
shouldn't happen if routes are correctly added, but makes sense anyway